ui/app/store/actions.js

```diff
@@ -2179,20 +2209,19 @@ export function setIpfsGateway(val) {
 export function updateCurrentLocale(key) {
   return async (dispatch) => {
     dispatch(showLoadingIndication())
-    await loadRelativeTimeFormatLocaleData(key)
-    return fetchLocale(key).then((localeMessages) => {
-      log.debug(`background.setCurrentLocale`)
-      background.setCurrentLocale(key, (err, textDirection) => {
-        if (err) {
-          dispatch(hideLoadingIndication())
-          dispatch(displayWarning(err.message))
-          return
-        }
-        switchDirection(textDirection)
-        dispatch(setCurrentLocale(key, localeMessages))
-        dispatch(hideLoadingIndication())
-      })
-    })
+
+    try {
+      await loadRelativeTimeFormatLocaleData(key)
+      const localeMessages = await fetchLocale(key)
+      const textDirection = await promisifiedBackground.setCurrentLocale(key)
```

Comment:
This looks like a change in behaviour. This function didn't block here previously.

Though... that doesn't seem good :thinking: This might be better.

Reply:
Ah, I meant to call this out! I tested it, and I'm not sure that there's a noticeable difference.